napari/layers/_shapes_layer/shape_list.py

```diff
@@ -565,3 +540,120 @@ def inside(self, coord):
+            String of shape shape_type, must be one of "{'line', 'rectangle',
+            'ellipse', 'path', 'polygon'}".
+
+        Returns
+        ----------
+        masks : np.ndarray
+            NxMxP array where the is one binary mask of shape MxP for each of
+            N shapes
+        """
+        if mask_shape is None:
+            mask_shape = self._vertices.max(axis=0).astype('int')
+
+        if shape_type is None:
+            data = [s.to_mask(mask_shape) for s in self.shapes]
+        elif shape_type not in self._types.keys():
+            raise ValueError("""shape_type not recognized, must be one of
+                         "{'line', 'rectangle', 'ellipse', 'path',
+                         'polygon'}"
+                         """)
+        else:
+            cls = self._types[shape_type]
+            data = ([s.to_mask(mask_shape) for s in self.shapes if
+                    type(s) == cls])
+        masks = np.array(data)
+
+        return masks
+
+    def to_labels(self, labels_shape=None, shape_type=None):
+        """Returns a integer labels image, where each shape is embedded in an
+        array of shape labels_shape with the value of the index + 1
+        corresponding to it, and 0 for background. Passing a `shape_type`
+        argument leads to only labels from that particular `shape_type` being
+        returned. These labels will be renumbered appropriately. For
+        overlapping shapes z-ordering will be respected.
+
+        Parameters
+        ----------
+        labels_shape : np.ndarray | tuple | None
+            1x2 array of shape of mask to be generated. If non specified, takes
+            the max of all the vertiecs
+        shape_type : str | None
+            String of shape shape_type, must be one of "{'line', 'rectangle',
+            'ellipse', 'path', 'polygon'}".
+
+        Returns
+        ----------
+        labels : np.ndarray
+            MxP integer array where the each value is either 0 for background
+            or an integer up to N for points inside the corresponding shape.
+        """
+        if labels_shape is None:
+            labels_shape = self._vertices.max(axis=0).astype('int')
+
+        labels = np.zeros(labels_shape, dtype=int)
+
+        if shape_type is None:
+            for ind in self._z_order[::-1]:
+                mask = self.shapes[ind].to_mask(labels_shape)
+                labels[mask] = ind+1
+        elif shape_type not in self._types.keys():
+            raise ValueError("""shape_type not recognized, must be one of
+                         "{'line', 'rectangle', 'ellipse', 'path',
+                         'polygon'}"
+                         """)
+        else:
+            cls = self._types[shape_type]
+            index = [1 if s == shape_type else 0 for s in self.shape_types]
+            index = np.cumsum(index)
+            for ind in self._z_order[::-1]:
+                shape = self.shapes[ind]
+                if type(shape) == cls:
```

Comment:
You usually want to use `isinstance` instead of a `type` comparison:
```suggestion
                if isinstance(shape, cls):
```

Reply:
fixed